pkg/kube/client.go

```diff
@@ -620,6 +621,41 @@ func (c *Client) waitForPodSuccess(obj runtime.Object, name string) (bool, error
 	return false, nil
 }
 
+// GetPodList uses the build in
+func (c *Client) GetPodList(namespace string, listOptions metav1.ListOptions) (*v1.PodList, error) {
+	podList, err := c.kubeClient.CoreV1().Pods(namespace).List(context.Background(), listOptions)
```

Comment:
What if users explicitly set the `metadata.namespace` parameter in their pods to some other namespace? We've seen that happen quite often with a few charts. Restricting to the namespace means we cannot fetch pods managed by the chart deployed to another namespace IIRC.

Reply:
It looks like this has been addressed now.. the deriveNamespace function is determining what gets passed in here.